yaml/landuse.yaml

```diff
@@ -305,7 +305,18 @@ filters:
   - filter: {leisure: water_park}
     min_zoom: GREATEST(LEAST(zoom, 16), 9)
     output: {kind: water_park}
-  - filter: {waterway: dam}
+  - filter:
+      waterway: dam
+      way:
+        expr: |
+          GeometryType(way) IN ('LINESTRING', 'MULTILINESTRING')
+    min_zoom: 12
+    output: {kind: dam}
+  - filter: 
+      waterway: dam
+      way:
+        expr: |
+          GeometryType(way) IN ('POLYGON', 'MULTIPOLYGON')
     min_zoom: GREATEST(LEAST(zoom, 16), 9)
     output: {kind: dam}
```

Comment:
I mentioned to use 2 rules the other day, but seeing this again with fresh eyes, do you think folding them together and having the condition worked out in the `expr` is clearer?

```yaml
  - filter: {waterway: dam}
    min_zoom: |
      CASE WHEN GeometryType(way) IN ('LINESTRING', 'MULTILINESTRING') THEN 12
           WHEN GeometryType(way) IN ('POLYGON', 'MULTIPOLYGON') THEN GREATEST(LEAST(zoom, 16), 9) END
    output: {kind: dam}
```